base/compiler/tfuncs.jl

```diff
@@ -1554,6 +1558,8 @@ function tuple_tfunc(argtypes::Vector{Any})
                 else
                     params[i] = Type
                 end
+            elseif !isvarargtype(x) && hasintersect(x, Type)
```

Comment:
I encountered a weird behavior here -- without this `!isvarargtype(x)` check `buildbot/xxx` builds fail with the following error:
```
Assertion failed: (!jl_is_vararg(x) && !jl_is_vararg(y)), function subtype, file /Users/julia/buildbot/worker/package_macos64/build/src/subtype.c, line 1261.
```
(e.g. https://build.julialang.org/#/builders/34/builds/4697)
while the same commit successfully builds w/o any problem on my machine (after `make cleanall` or even a fresh clone).
Maybe buildbot uses a different method to build something, and in that case the bootstrap makes more use of interpreter or something?

Reply:
Turned out that I needed to enable ["assert build"](https://github.com/JuliaLang/julia/blob/master/doc/src/devdocs/build/build.md#building-an-assert-build-of-julia).